src/main/java/io/gravitee/identityprovider/api/social/SocialAuthenticationProvider.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.identityprovider.api.social;
+
+import io.gravitee.common.component.Lifecycle;
+import io.gravitee.identityprovider.api.Authentication;
+import io.gravitee.identityprovider.api.AuthenticationProvider;
+import io.gravitee.identityprovider.api.User;
+import io.reactivex.Maybe;
+import io.reactivex.Single;
+
+/**
+ * @author Lorie PISICCHIO (lorie.pisicchio at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public interface SocialAuthenticationProvider extends AuthenticationProvider {
+
+    Single<String> signInUrl();
+
+    Maybe<User> loadUserByUsername(Authentication authentication);
+
+    String getRedirectUrl();
+
+    Single<SocialAuthenticationProviderConfiguration> getIntegrationConfiguration();
```

Comment:
Why not just `getConfiguration()` instead of `getIntegrationConfiguration()` ?

Reply:
Because there is already a getConfiguration in AbstractSocialAuthenticationProvider used to retreive the backend config for a social auth provider. This one is to expose configuration for frontend. https://github.com/gravitee-io/gravitee-cockpit/blob/feat/%2367-social-authent/gravitee-cockpit-idp/gravitee-cockpit-idp-common/src/main/java/com/graviteesource/cockpit/idp/common/AbstractSocialAuthenticationProvider.java#L46
But maybe we could rename it more explicitly or find a way to avoid method name collision ?

Reply:
Ok got it now! So I suggest to completely remove this method, move the `SocialAuthenticationProviderConfiguration` class somewhere else and let application to manage all alone.
That means that if you have a dedicated service in charge of loading the configuration of social providers, it should certainly do the job (eg: resolve the sign in url, populate other configuration attributes).